src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/DynamicInvokeInfo.cs

```diff
@@ -769,7 +769,7 @@ private unsafe void CopyBackToArray(ref object? src, object?[] dest)
                         Debug.Assert(type.IsPointer);
                         obj = Pointer.Box((void*)Unsafe.As<byte, IntPtr>(ref obj.GetRawData()), type);
                     }
-                    if ((transform & Transform.FunctionPointer) != 0)
+                    else if ((transform & Transform.FunctionPointer) != 0)
```

Comment:
Nit: This would produce a bit smaller code:
```
else
{
   obj = RuntimeImports.RhBox(
       ((transform & Transform.FunctionPointer) != 0) ? EETypePtr.EETypePtrOf<IntPtr>() : argumentInfo.Type,
       ref obj.GetRawData());
}
```